src/commands/runPrompt.ts

```diff
@@ -168,42 +168,54 @@ export const runPrompt: (secrets: vscode.SecretStorage, mode: PromptOption) => v
         await spawnPromptImage(promptOption.id, runningLocal ? inputWorkspace! : workspaceFolder!.uri.fsPath, 'vscode-user', process.platform, async (json) => {
-            if (json.method === 'functions') {
-                const functions = json.params;
-                for (const func of functions) {
-                    const { id, function: { arguments: args } } = func;
-                    const params_str = args;
-                    let functionRange = ranges[id] || getBaseFunctionRange();
-                    if (functionRange.isSingleLine) {
-                        // Add function to the end of the file and update the range
-                        await writeToEditor(params_str);
-                        functionRange = new vscode.Range(functionRange.start.line, functionRange.start.character, doc.lineCount, 0);
+            switch (json.method) {
+                case 'functions':
+                    const functions = json.params;
+                    for (const func of functions) {
+                        const { id, function: { arguments: args } } = func;
+                        const params_str = args;
+                        let functionRange = ranges[id] || getBaseFunctionRange();
+                        if (functionRange.isSingleLine) {
+                            // Add function to the end of the file and update the range
+                            await writeToEditor(params_str);
+                            functionRange = new vscode.Range(functionRange.start.line, functionRange.start.character, doc.lineCount, 0);
+                        }
+                        else {
+                            // Replace existing function and update the range
+                            await writeToEditor(params_str, functionRange);
+                            functionRange = new vscode.Range(functionRange.start.line, functionRange.start.character, functionRange.end.line + params_str.split('\n').length, 0);
+                        }
+                        ranges[id] = functionRange;
                     }
-                    else {
-                        // Replace existing function and update the range
-                        await writeToEditor(params_str, functionRange);
-                        functionRange = new vscode.Range(functionRange.start.line, functionRange.start.character, functionRange.end.line + params_str.split('\n').length, 0);
+                    break;
+                case 'start':
+                    const { level, role, content } = json.params;
+                    const header = Array(level).fill('#').join('');
+                    await writeToEditor(`${header} ROLE ${role} (${content})\n`);
+                    break;
+                case 'functions-done':
+                    await writeToEditor(json.params.content);
+                    break;
+                case 'message':
+                    await writeToEditor(json.params.content);
+                    if (json.params.debug) {
```

Comment:
I feel like this can't be right because `message` notifications can also have `content`.  And the `debug` messages should only be shown when we're in debug mode.  Otherwise, this is just streaming content from the assistant.  I haven't tried this but it feels like it must not work just looking at the code.

Reply:
I haven't implemented the debug mode or colors yet.